### ISSUE TITLE
Guest link variable in mails returns wrong link

### ISSUE THREAD

**Issue by linusgke** [osTicket/osTicket#6491]
Hey there,

I recently encountered an issue where the variable used for displaying the guest access link in a mail template turned out to be an internal access link. 

%{recipient.ticket_link}
returned ".../scp/tickets.php?id=949"
for a mail sent to a non-agent (guest).

This happens, whenever the mail addresses of people who are also agents in the systems are added to the list of associated people of a ticket.

Problem herewith is, that the guest also receives the mail with the internal access link and is therefore forced to generate a new access link. This procedure seems to be too much of a challenge especially for unskilled users.

Would it be possible to fix this issue, or is there anything I can do (e.g. use another variable) to avoid this issue?

Fixing meaning: Sending the guest link to guests and the internal link to agents instead of sending everybody the internal link when using said variable.

Thanks in advance!

Kind regards
Linus

**Comment by JediKev** [comment]
@elektr0nisch 

It sounds like your **Helpdesk URL** (**Admin Panel > Settings > System**) is incorrect or the email template has `/scp` hardcoded in it. I would double check to make sure none of these include `/scp`.

Cheers.

**Comment by linusgke** [comment]
The follwoing is configured in the mail template:

![grafik](https://user-images.githubusercontent.com/38865194/225035317-67726b17-db80-445c-bdcb-0eb0622447e7.png)

The Helpdesk URL is also specified correctly.

![grafik](https://user-images.githubusercontent.com/38865194/225035651-34a8b56f-faba-468b-9a9d-db2271498382.png)

Which is only logical, since the link sent to agents works perfectly, since they're logged in. But when a non-agent user is trying to open said link, it'll prompt them with the agent login screen. Keep in mind that this error only occurs when adding an agent's mail address as CC (assosicated) in a ticket. As soon as this is done, also non-agent users will receive an email with the agent only (internal) link. Guest users should receive an auth-link looking so: `https://example.com/view.php?auth=<token>` not an `https://example.com/scp/tickets.php?id=<id>` link.

![grafik](https://user-images.githubusercontent.com/38865194/225036828-889ab9b0-6e22-4977-a410-d4b3b3eb09e8.png)

(Sorry, our system is using German!)

Would be glad if this can be resolved.
Thanks in advance!

Kinds regards
Linus

**Comment by JediKev** [comment]
@elektr0nisch 

Can you post a screenshot of **Admin Panel > Dashboard > Information**?

Cheers.

**Comment by linusgke** [comment]
Sure thing!

![grafik](https://user-images.githubusercontent.com/38865194/225040123-dd6381fe-be19-4585-b0a8-e677c6cc1150.png)

**Comment by JediKev** [comment]
@elektr0nisch 

Okay, can you please upgrade to v1.17.3 and retest? If you can still replicate it using the latest version then we will do further debugging and attempt to replicate it on our end.

Cheers.

**Comment by linusgke** [comment]
Alright! Be right back :)

Thank you!

**Comment by linusgke** [comment]
Successfully upgraded to v1.17.3 and retested. Problem still occurs!

#### What did I do?

![grafik](https://user-images.githubusercontent.com/38865194/225047201-96313af1-d0c1-4e5e-bfb4-89cbeb781c1f.png)
![grafik](https://user-images.githubusercontent.com/38865194/225047344-ed5436f5-45a8-4a69-addb-5e136e24c4ab.png)
Created ticket as "linus@elektronisch.dev" (our guest) and added "linus.groschke@waldorf-augsburg.de" (my agent email) as CC.

Now whenever new activity is reported to the guest by mail, the variable `%{recipient.ticket_link}` returns the following link:
![grafik](https://user-images.githubusercontent.com/38865194/225046763-ab79da13-4874-4c81-86be-b12fcc8f967f.png)

Which, of course is not accessible for the guest.

**Comment by linusgke** [comment]
Brief update on this: It doesn't matter if the person specified in CC is an agent or not. Once a person is specified in CC, all email recipients will receive the link with "view.php" instead of an auth link.

**Comment by JediKev** [comment]
@elektr0nisch 

So to be clear, you're saying that after upgrading you no longer get the `/scp` link however you do not get an `auth=` string added, correct? Just trying to see if you are still seeing `/scp` in the links when Agent is CC.

Cheers.

**Comment by linusgke** [comment]
Sorry, it's been a bit confusing to me either... I falsely accused the system of adding `/scp` to the guest access-link. This, as I have tried on another instance still running the older version, did never happen. The actual problem is, that as soon as anybody gets added as CC, all people receiving notifcation mails for that ticket, will get a `domain.tld/view.php?id=<id>` type url, instead of an `auth=` type url, which they would need to obtain immediate access.

**Comment by JediKev** [comment]
@elektr0nisch 

Okay, that makes _much_ more sense now. So this is actually intended behavior of the system. When we generate the alert we send a single email with all CCs added. If there are ANY number of CCs on a Ticket then we will not add the auth string. Reason being is how would we know which auth string to use? Also, if we do add one then the auth string would be for one particular user and every other user that clicks that would be logged in as the incorrect account.

TLDR; If there are CCs on a Ticket the recipient_link will not include an auth string.

Cheers.

**Comment by linusgke** [comment]
Okay, that makes a lot more sense! Wouldn't it in theory be possible, to send individual mails instead of adding the CCs of the ticket as CCs of the email. How am I supposed to grant CCed users access to the full ticket correspondance?
